node/src/syncer.rs

```diff
@@ -618,12 +613,28 @@ where
 
     let network_head = p2p.get_head_header().await?;
 
-    // Insert HEAD to the store and initialize header-sub
+    // Insert HEAD to the store and initialize header-sub.
+    // This will apply insertion restrictions.
     store.insert(network_head.clone()).await?;
 
     Ok(network_head)
 }
 
+async fn header_sub_recv(
+    rx: &mut Option<mpsc::Receiver<ExtendedHeader>>,
```

Comment:
I think it'd be more idiomatic to accept `Option<&mut ...>`

Reply:
done